jaegertracing/jaeger
# exp: Support primary and archive storage

## Which problem is this PR solving?
- Part of #4843
- Prototype defining primary and archive storage used by query service

## Description of the changes
- Introduce `trace_storage_archive` argument to extension/jaegerquery
  - This is temporary, need to design a better shape for the config
- Implement initialization of archive storage. Tested manually.
- Changed memstore to implement archive storage (maybe not a good idea in retrospect)
- Change how no-config condition is detected as the previous changes worked for all-in-one but broke with-config mode
  - Need to start adding some basic integration tests
  - There is a better way than using default configs for all in one - override `--config` flag with `yaml: ....` string an provide configuration as a file (embed the file into the binary)
- Some refactoring of querysvc/app to be able to reuse querysvc.QueryOptions in v2 config

## How was this change tested?
- Run all-in-one
- Run with config
- Run with custom UI config to enable Archive button (it should really be dynamically determined by backend capability)
